src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -210,18 +280,12 @@ public ScopeData(Action enter = null, Action exit = null)
 
             public void Enter()
             {
-                if (_enter != null)
-                {
-                    _enter();
-                }
+                _enter?.Invoke();
```

Comment:
please remove the null checks in here - we will be paying relatively high cost for that where this is only needed for chip select and we can pass no op for that case

Reply:
The code was like this, I just changed it to ? instead of `if`. But I agree, I will remove the check completely.